sdk/keyvault/azure-security-test-keyvault-jca/src/test/java/com/azure/security/keyvault/jca/KeyVaultKeyStoreTest.java

```diff
@@ -0,0 +1,179 @@
+        + "9w0BAQsFAAOCAQEAGp8mCioVCmM+kZv6r+K2j2uog1k4HBwN1NfRoSsibDB8+QXF"
+        + "bmNf3M0imiuR/KJgODyuROwaa/AalxNFMOP8XTL2YmP7XsddBs9ONHHQXKjY/Ojl"
+        + "PsIPR7vZjwYPfEB+XEKl2fOIxDQQ921POBV7M6DdTC49T5X+FsLR1AIIfinVetT9"
+        + "QmNuvzulBX0T0rea/qpcPK4HTj7ToyImOaf8sXRv2s2ODLUrKWu5hhTNH2l6RIkQ"
+        + "U/aIAdQRfDaSE9jhtcVu5d5kCgBs7nz5AzeCisDPo5zIt4Mxej3iVaAJ79oEbHOE"
+        + "p192KLXLV/pscA4Wgb+PJ8AAEa5B6xq8p9JO+Q==";
+
+    private KeyVaultKeyStore keystore;
+
+    private String certificateName;
+
+    public static void putEnvironmentPropertyToSystemProperty(List<String> key) {
+        key.forEach(
+            environmentPropertyKey -> {
+                String value = System.getenv(environmentPropertyKey);
+                if (value != null) {
+                    String systemPropertyKey = environmentPropertyKey.toLowerCase().replaceFirst("azure_keyvault_",
+                        "azure.keyvault.").replaceAll("_", "-");
+                    System.getProperties().put(systemPropertyKey, value);
+                }
+            }
+        );
+    }
+
+    @BeforeEach
+    public void setEnvironmentProperty() {
+        KeyVaultLoadStoreParameter parameter = new KeyVaultLoadStoreParameter(
+            System.getenv("AZURE_KEYVAULT_URI"),
+            System.getenv("AZURE_KEYVAULT_TENANT_ID"),
+            System.getenv("AZURE_KEYVAULT_CLIENT_ID"),
+            System.getenv("AZURE_KEYVAULT_CLIENT_SECRET"));
+        certificateName = System.getenv("AZURE_KEYVAULT_CERTIFICATE_NAME");
+        putEnvironmentPropertyToSystemProperty(
+            Arrays.asList("AZURE_KEYVAULT_URI",
+                "AZURE_KEYVAULT_TENANT_ID",
+                "azure.keyvault.aad-authentication-url",
+                "AZURE_KEYVAULT_CLIENT_ID",
+                "AZURE_KEYVAULT_CLIENT_SECRET")
+        );
+        keystore = new KeyVaultKeyStore();
+        keystore.engineLoad(parameter);
+    }
+
+    @Test
+    public void testEngineGetCertificate() {
+        assertNotNull(keystore.engineGetCertificate(certificateName));
+    }
+
+    @Test
+    public void testEngineGetCertificateAlias() {
+        X509Certificate certificate;
+
+        try {
+            byte[] certificateBytes = Base64.getDecoder().decode(TEST_CERTIFICATE);
+            CertificateFactory cf = CertificateFactory.getInstance("X.509");
+            certificate = (X509Certificate) cf.generateCertificate(new ByteArrayInputStream(certificateBytes));
+        } catch (CertificateException e) {
+            throw new ProviderException(e);
+        }
+        keystore.engineSetCertificateEntry("setcert", certificate);
+        assertNotNull(keystore.engineGetCertificateAlias(certificate));
+    }
+
+    @Test
+    public void testEngineGetCertificateChain() {
+        assertNotNull(keystore.engineGetCertificateChain(certificateName));
+    }
+
+
+    @Test
+    public void testEngineSetCertificateEntry() {
```

Comment:
Same for this test. And for the common code block, what about we write another function for it?